platform/android/java/lib/src/org/godotengine/godot/input/GodotInputHandler.java

```diff
@@ -366,4 +424,67 @@ private int findJoystickDevice(int device_id) {
 
 		return -1;
 	}
+
+	private boolean handleMouseEvent(final MotionEvent event) {
+		switch (event.getActionMasked()) {
+			case MotionEvent.ACTION_HOVER_ENTER:
+			case MotionEvent.ACTION_HOVER_MOVE:
+			case MotionEvent.ACTION_HOVER_EXIT: {
+				final float x = event.getX();
+				final float y = event.getY();
+				final int type = event.getAction();
+				queueEvent(new Runnable() {
+					@Override
+					public void run() {
+						GodotLib.hover(type, x, y);
+					}
+				});
+				return true;
+			}
+			case MotionEvent.ACTION_BUTTON_PRESS:
+			case MotionEvent.ACTION_BUTTON_RELEASE:
+			case MotionEvent.ACTION_MOVE: {
+				final float x = event.getX();
+				final float y = event.getY();
+				final int buttonsMask = InputUtils.fixButtonsStateMask(event.getButtonState());
+				final int action = event.getAction();
+				queueEvent(new Runnable() {
+					@Override
+					public void run() {
+						GodotLib.touch(event.getSource(), action, 0, 1, new float[] { 0, x, y }, buttonsMask, 1);
+					}
+				});
+				return true;
+			}
+			case MotionEvent.ACTION_SCROLL: {
+				final float[] positionArray = new float[] { 0, event.getX(), event.getY() };
+				final int buttonsMaskFixed = InputUtils.fixButtonsStateMask(event.getButtonState());
+
+				if (event.getAxisValue(MotionEvent.AXIS_VSCROLL) > 0.0f) {
+					handleMouseScrollWheelButtonEvent(event.getSource(), buttonsMaskFixed, MOUSE_WHEEL_UP, event.getAxisValue(MotionEvent.AXIS_VSCROLL), positionArray);
+				} else if (event.getAxisValue(MotionEvent.AXIS_VSCROLL) < 0.0f) {
```

Comment:
Seems this can be a regular `else` instead of an `else if`.

Reply:
I forgot why I did that, I need that in case of right or left axis, and than vscroll can be 0 so we don't want to trigger down scroll

Reply:
@thebestnom If `vscroll` is `0`, shouldn't that be propagated via the `factor` argument to the native code, thus preventing any down scroll since the `factor` would also be `0`?

Reply:
It will be prevented scroll, but it also fire an event (with 0 factor) and I don't think firing an 0 scroll event make sense...

Reply:
Sounds good, thanks for the clarification!

Reply:
@thebestnom Have you also tested that `diagonal` scrolls are handled then, for example for mouse that allows to click the middle button and scrolls based on where the pointer is moved. Or is that handled by another logic.

Reply:
I tested that once, but I don't have that mouse anymore, but the logic should just work (TM)

Reply:
In all seriousness, it does trigger 2 events when I changed to else instead, so in that regard it does work

Reply:
Sounds good!